### ISSUE TITLE
Getting Started with DataBricks

### ISSUE THREAD

**Issue by Misbah952** [amjadraza/spark-ml-course#9]
This article introduces the set of fundamental concepts you need to understand in order to use Databricks Workspace effectively.

**Workspace**
A workspace is an environment for accessing all of your Databricks assets. A workspace organizes objects (notebooks, libraries, dashboards, and experiments) into folders and provides access to data objects and computational resources.

This section describes the objects contained in the Databricks workspace folders.

**Notebook**

A web-based interface to documents that contain runnable commands, visualizations, and narrative text.

**Dashboard**

An interface that provides organized access to visualizations.

**Library**

A package of code is available to the notebook or job running on your cluster. Databricks runtimes include many libraries and you can add your own.

**Repo**

A folder whose contents are co-versioned together by syncing them to a remote Git repository.

**Experiment**

A collection of MLflow runs for training a machine learning model.

**Interface**
This section describes the interfaces that Databricks supports for accessing your assets: UI, API, and command-line (CLI).

**UI**

The Databricks UI provides an easy-to-use graphical interface to workspace folders and their contained objects, data objects, and computational resources.

![image](https://user-images.githubusercontent.com/80267029/130975017-b6da34ba-532d-435e-865c-b9e0ec050b45.png)

**REST API**

There are two versions of the REST API: REST API 2.0 and REST API 1.2. The REST API 2.0 supports most of the functionality of the REST API 1.2, as well as additional functionality and is preferred.

**CLI**

An open-source project hosted on GitHub. The CLI is built on top of the REST API 2.0.

**Data management**

This section describes the objects that hold the data on which you perform analytics and feed into machine learning algorithms.

**Databricks File System (DBFS)**

A filesystem abstraction layer over a blob store. It contains directories, which can contain files (data files, libraries, and images), and other directories. DBFS is automatically populated with some datasets that you can use to learn Databricks.

**Database**

A collection of information that is organized so that it can be easily accessed, managed, and updated.

**Table**

A representation of structured data. You query tables with Apache Spark SQL and Apache Spark APIs.

**Metastore**

The component that stores all the structure information of the various tables and partitions in the data warehouse including column and column type information, the serializers and deserializers necessary to read and write data, and the corresponding files where the data is stored. Every Databricks deployment has a central Hive metastore accessible by all clusters to persist table metadata. You also have the option to use an existing external Hive metastore.

**Computation management**

This section describes concepts that you need to know to run computations in Databricks.

**Cluster**

A set of computation resources and configurations on which you run notebooks and jobs. There are two types of clusters: all-purpose and job.

You create an all-purpose cluster using the UI, CLI, or REST API. You can manually terminate and restart an all-purpose cluster. Multiple users can share such clusters to do collaborative interactive analysis.
The Databricks job scheduler creates a job cluster when you run a job on a new job cluster and terminates the cluster when the job is complete. You cannot restart a job cluster.
Pool

A set of idle, ready-to-use instances that reduce cluster start and auto-scaling times. When attached to a pool, a cluster allocates its driver and worker nodes from the pool. If the pool does not have sufficient idle resources to accommodate the cluster’s request, the pool expands by allocating new instances from the instance provider. When an attached cluster is terminated, the instances it used are returned to the pool and can be reused by a different cluster.

**Databricks runtime**

The set of core components that run on the clusters managed by Databricks. Databricks offers several types of runtimes:

**Databricks Runtime** includes Apache Spark but also adds a number of components and updates that substantially improve the usability, performance, and security of big data analytics.
**Databricks Runtime for Machine Learning** is built on Databricks Runtime and provides a ready-to-go environment for machine learning and data science. It contains multiple popular libraries, including TensorFlow, Keras, PyTorch, and XGBoost.
b is a version of Databricks Runtime optimized for working with genomic and biomedical data.
**Databricks Light** is the Databricks packaging of the open-source Apache Spark runtime. It provides a runtime option for jobs that don’t need the advanced performance, reliability, or autoscaling benefits provided by Databricks Runtime. You can select Databricks Light only when you create a cluster to run a JAR, Python, or spark-submit job; you cannot select this runtime for clusters on which you run interactive or notebook job workloads.
Job

A non-interactive mechanism for running a notebook or library either immediately or on a scheduled basis.

**Workload**

Databricks identifies two types of workloads subject to different pricing schemes: data engineering (job) and data analytics (all-purpose).

Data engineering An (automated) workload runs on a job cluster which the Databricks job scheduler creates for each workload.
Data analytics An (interactive) workload runs on an all-purpose cluster. Interactive workloads typically run commands within a Databricks notebook. However, running a job on an existing all-purpose cluster is also treated as an interactive workload.
Execution context

The state for a REPL environment for each supported programming language. The languages supported are Python, R, Scala, and SQL.

**Machine learning**
This section describes concepts related to machine learning in Databricks.

**Experiments**

The main unit of organization for tracking machine learning model development. Experiments organize, display, and control access to individual logged runs of model training code.

**Feature Store**

A centralized repository of features. Databricks Feature Store enables feature sharing and discovery across your organization and also ensures that the same feature computation code is used for model training and inference.

**Models**

A trained machine learning or deep learning model that has been registered in Model Registry.

**Authentication and Authorization**
This section describes concepts that you need to know when you manage Databricks users and their access to Databricks assets.

**User**

A unique individual who has access to the system.

**Group**

A collection of users.

**Access control list (ACL)**

A list of permissions is attached to the workspace, cluster, job, table, or experiment. An ACL specifies which users or system processes are granted access to the objects, as well as what operations are allowed on the assets. Each entry in a typical ACL specifies a subject and an operation.

**Launch the signup wizard and select a subscription type**
Click Try [Databricks](https://databricks.com/try-databricks?_ga=2.210615760.459890236.1629985428-1257345202.1629827668) here or at the top of this page.

Enter your name, company, email, and title, and click GET STARTED FOR FREE.

Select the free program you want:

For Community Edition, click the GET STARTED button.
For a 14-day free trial, select the cloud provider you want to use: Azure, AWS, or Google Cloud. The Databricks trial is free, but you must have a cloud account, and Databricks uses compute and storage resources in your cloud account. Each cloud provider has programs for free credits during your free trial.
![image](https://user-images.githubusercontent.com/80267029/130976172-45e056cf-4000-4236-85c0-3e5e18c558ba.png)
Look for an email asking you to verify your email address.

Follow the instructions for your selection: Databricks Platform Free Trial or Community Edition.

**Databricks Platform Free Trial**
After you’ve signed up for a free trial, you’ll see a page announcing that an email has been sent to the address you provided. To learn what to do next, follow the instructions in Set up your Databricks account and deploy a workspace.

**Note**

The Databricks trial is free, but you must have an AWS account, and Databricks uses compute and S3 storage resources in your AWS account. If you want to request free AWS credits, go to the Proof of Concept Credit Available page. Fill out and submit the PoC credit application and wait until you are contacted by Databricks before you continue with the trial registration process.

**Community Edition**
After you’ve signed up for a Community Edition account, you’ll see a page announcing that an email has been sent to the address you provided.

Open the Welcome Email you received from Databricks after you signed up for an account, and click the link to verify your email address and create your password.

When you click Submit, you’ll be taken to the Databricks Community Edition home page.

![image](https://user-images.githubusercontent.com/80267029/130976393-494431ae-bee7-4fff-a377-97ed1edc7703.png)

**Step 1: Orient yourself to the Databricks Data Science & Engineering UI**
![image](https://user-images.githubusercontent.com/80267029/130976826-0b99171f-0d4e-40be-9f26-8e5b8ca95ccd.png)

From the sidebar at the left and the Common Tasks list on the landing page, you access fundamental Databricks Data Science & Engineering entities: the Workspace, clusters, tables, notebooks, jobs, and libraries. The Workspace is the special root folder that stores your Databricks assets, such as notebooks and libraries, and the data that you import.

**Use the sidebar**
You can access all of your Databricks assets using the sidebar. The sidebar’s contents depend on the selected persona: Data Science & Engineering, Machine Learning, or SQL.

- By default, the sidebar appears in a collapsed state and only the icons are visible. Move your cursor over the sidebar to expand to the full view.
- 
- To change the persona, click the icon below the![image](https://user-images.githubusercontent.com/80267029/130977026-67df49f5-a59d-4d92-9510-72d42aa334b0.png) Databricks logo, and select a persona.

![image](https://user-images.githubusercontent.com/80267029/130977093-535da441-0a27-4405-8130-ae82021bc663.png)

- To pin a persona so that it appears next time you log in, click 
 ![image](https://user-images.githubusercontent.com/80267029/130977541-a91f8fbe-25d3-4c20-914c-e2e8ab45d250.png)
  next to the persona. Click it again to remove the pin.

- Use Menu options at the bottom of the sidebar to set the sidebar mode to Auto (default behavior), Expand, or Collapse.

- When you open a machine learning-related page, the persona automatically switches to Machine Learning.

**Get help**
To get help, click the question icon 
![image](https://user-images.githubusercontent.com/80267029/130977754-fe1cc7ff-aaaf-4279-8932-08bbd84147c5.png)
 at the top right-hand corner.

![image](https://user-images.githubusercontent.com/80267029/130977800-89f7541d-8cf2-4776-b0ad-fffd2f69d0e2.png)

**Step 2: Create a cluster**

A cluster is a collection of Databricks computation resources. To create a cluster:

1. In the sidebar, click 
![image](https://user-images.githubusercontent.com/80267029/130977960-4b7c44a6-9e15-49a5-bb98-ecfe4ad86640.png)
Compute.

2. On the Compute page, click Create Cluster.

![image](https://user-images.githubusercontent.com/80267029/130978063-b07cf974-fe4d-403b-8698-bf8ee6e998eb.png)

3. On the Create Cluster page, specify the cluster name Quickstart and select 7.3 LTS (Scala 2.12, Spark 3.0.1) in the Databricks Runtime Version drop-down.

4. Click Create Cluster.

**Step 3: Create a notebook**
A notebook is a collection of cells that run computations on an Apache Spark cluster. To create a notebook in the Workspace:

1. In the sidebar, click 
![image](https://user-images.githubusercontent.com/80267029/130978357-3485cbd1-ed94-49e9-92cf-bb4bc21c194c.png)
Workspace.
2. In the Workspace folder, select Down Caret Create > Notebook.

![image](https://user-images.githubusercontent.com/80267029/130978420-4da30d7a-8979-4fe5-88b0-b822e7d03036.png)

3. On the Create Notebook dialog, enter a name and select SQL in the Language drop-down. This selection determines the default language of the notebook.

4. Click Create. The notebook opens with an empty cell at the top.
